drivers/sd-card/sd.c

```diff
@@ -117,19 +125,23 @@ static int32_t wait_for_response(struct sd_desc *sd_desc, uint8_t *data_out)
  */
 static int32_t wait_until_not_busy(struct sd_desc *sd_desc)
 {
-	uint8_t	data;
-	uint32_t timeout = WAIT_RESP_TIMEOUT;
+	uint32_t	not_timeout;
+	int32_t		ret;
+	uint8_t		data;
 
-	data = 0x00;
-	while (data == 0x00) {
+	ret = FAILURE;
+	not_timeout = WAIT_RESP_TIMEOUT;
+	do {
 		data = 0xFF;
 		if (SUCCESS != spi_write_and_read(sd_desc->spi_desc, &data, 1))
-			return FAILURE;
-		if (timeout-- == 0)
-			return FAILURE;
-	}
+			break;
+		if (data != 0x00) {
+			ret = SUCCESS;
+			break;
+		}
+	} while (not_timeout--);
```

Comment:
Should we have an `mdelay` in this while too?

Reply:
Yes we should, I missed it